hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -584,17 +766,45 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
                     f"Maker sell order ({limit_order_record.quantity} {limit_order_record.base_currency} @ "
                     f"{limit_order_record.price} {limit_order_record.quote_currency}) is filled."
                 )
-            else:
+                # Leftover other side maker order will be left in the market until its expiration or potential fill
+                # Since the sell side side was filled, the buy side maker order is unlikely to be filled, therefore
+                # it'll likey expire
+                # The others are left in the market to collect market making fees
+                # Meanwhile new maker order may be placed
+            if order_id in self._taker_to_maker_order_ids.keys():
                 self.log_with_clock(
                     logging.INFO,
                     f"({market_pair.taker.trading_pair}) Taker sell order {order_id} for "
-                    f"({order_completed_event.base_asset_amount} {order_completed_event.base_asset} has been completely filled."
+                    f"({order_completed_event.base_asset_amount} {order_completed_event.base_asset} "
+                    f"has been completely filled."
                 )
                 self.notify_hb_app_with_timestamp(
-                    f"Taker sell order {order_completed_event.base_asset_amount} {order_completed_event.base_asset} is filled."
+                    f"Taker SELL order ({order_completed_event.base_asset_amount} {order_completed_event.base_asset} "
+                    f"{order_completed_event.quote_asset}) is filled."
                 )
+                # Cleanup maker fill events - no longer needed to create taker orders
+                del self._order_fill_buy_events[market_pair]
 
```

Comment:
In the case of multiple taker orders, this would cause other taker orders to be ignored and not re-placed if needed

Reply:
Checking if it's the last taker order should be done before this, I believe

Reply:
✅